test/adaptive_load/fake_time_source.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include "envoy/common/time.h"
+
+namespace Nighthawk {
+/**
+ * Fake time source that ticks 1 second on every query, starting from the Unix epoch. Supports only
+ * monotonicTime().
+ */
+class FakeIncrementingMonotonicTimeSource : public Envoy::TimeSource {
```

Comment:
I thought envoy already supported mock timers.  Is there a reason why we need to implement a fake?

Reply:
I see a mock TimeSource being used, and there's TestTimeSystem, but here we need something where:
- time advances between calls
- no intervention is needed between calls

because once we unleash the main loop we can't pause it to manipulate the clock.

Reply:
So personally, I love the simplicity of this. It might be possible to use simulated time, but then we may need another thread to advance time or some other relatively complex construct. Also, as I think monotonic time promises us to advance each time we call it, this seems quite an accurate modelling of it. So personally, when weighing the potential duplication introduced against the simplicity we win, I'm rooting for the way this is right now.